kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/encrypt/InBandEncryptionManager.java

```diff
@@ -68,7 +72,7 @@ public InBandEncryptionManager(@NonNull Encryption encryption,
     }
 
     @VisibleForTesting
-    public CompletionStage<Dek<E>> currentDek(@NonNull EncryptionScheme<K> encryptionScheme) {
+    public CompletionStage<EncryptionDekCache<K, E>.EncryptionDek> currentDek(@NonNull EncryptionScheme<K> encryptionScheme) {
```

Comment:
easy enough to fix up by making it a static class instead of a member

Reply:
have extracted a `DekContext` interface which only needs to know `<E>`. The implementation needs to be aware of `<K>` so that it can invalidate the cache, but we can hide that from clients.